src/main/java/dk/kb/labsapi/ImageExport.java

```diff
@@ -0,0 +1,392 @@
+
+        return String.format(Locale.ROOT, "&RGN=%1.5f,%1.5f,%1.5f,%1.5f", calculatedX, calculatedY, calculatedW, calculatedH);
+    }
+
+    /**
+     * Download an illustration from given URL and return it as a byte array.
+     * @param url pointing to the image to download.
+     * @return downloaded image as byte array.
+     */
+    private byte[] downloadSingleIllustration(URL url) {
+        /*
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        byte[] illustrationAsByteArray = new byte[0];
+        try (InputStream is = url.openStream()) {
+            byte[] bytes = new byte[4096];
+            int n;
+
+            while ((n = is.read(bytes)) > 0) {
+                baos.write(bytes, 0, n);
+            }
+            illustrationAsByteArray = baos.toByteArray();
+        } catch (IOException e) {
+            log.error("Failed to download illustration from " + url + " while reading bytes");
+        }
+        return illustrationAsByteArray;
+         */
+        try {
+            return IOUtils.toByteArray(url);
+        }  catch (IOException e) {
+            log.error("Failed to download illustration from " + url + " while reading bytes");
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Streams content of all URLs given in input list to an output stream that delivers a zip file of all images from the URLs.
+     * @param illustrationURLs List of URls to get content from.
+     * @param illustrationMetadata List of metadata for each image returned from the URL list. Used to construct filenames.
+     * @param output output stream which holds the outputted zip file.
+     */
+    private void illustrationURLSToStream(List<URL> illustrationURLs, List<IllustrationMetadata> illustrationMetadata, OutputStream output, Map<String, Object> metadataMap) throws IOException {
+        ZipOutputStream zos = new ZipOutputStream(output);
+        zos.setLevel(Deflater.NO_COMPRESSION);
+
+        // Add metadata file to zip
+        try {
+            addMetadataFileToZip(metadataMap, zos);
+        } catch (Exception e) {
+            String message = String.format(
+                    Locale.ROOT,
+                    "Exception adding metadata entry to ZIP with %s illustrationURLs and %s illustrationMetadatas",
+                    illustrationURLs == null ? "null" : illustrationURLs.size(),
+                    illustrationMetadata == null ? "null" : illustrationMetadata.size());
+            log.warn(message, e);
+            throw new IOException(message, e);
+        }
+
+        int count = 0;
+        try {
+            for (int i = 0; i < illustrationURLs.size() ; i++) {
+                byte[] illustration = downloadSingleIllustration(illustrationURLs.get(i));
+                String pageUuid = illustrationMetadata.get(i).getPageUUID();
+                addToZipStream(illustration, String.format(Locale.ROOT, "pageUUID_%s_illustration_%3d.jpeg", pageUuid, count), zos);
```

Comment:
This causes filenames with spaces. Could you add leading zeroes to the counter to avoid that? `pageUUID_%s_illustration_%03d.jpeg`.

Very personal preference: `jpg`, not `jpeg`. Let's ask the others.